### ISSUE TITLE
Jet/Hera: internal compiler error w/ atmosphere.F90

### ISSUE THREAD

**Issue by MicroTed** [ufs-community/ufs-weather-model#1030]
## Description

I was trying to compile the current develop branch of ufs on jet, and it gets to 97% done and has an internal compiler error (same on hera). I'm using the ufs_jet.intel (ufs_hera.intel) module. (Nice that the hera module also sets up miniconda -- that would be good to have in the jet module, too.) [Same result for DEBUG=ON and ufs_jet.intel_debug]  And I have

CCPP_SUITES=FV3_GFS_v15p2
CMAKE_FLAGS=-DAPP=ATM

Here's the error:

cd /lfs4/NAGAPE/hpc-wof1/mansell/ufs/ufs-100-lr/src/ufs-stock/build/FV3/atmos_cubed_sphere && /apps/intel/compilers_and_libraries_2018.5.274/linux/mpi/intel64/bin/mpiifort -DENABLE_QUAD_PRECISION -DGFS_PHYS -DGFS_TYPES -DINTERNAL_FILE_NML -DMOIST_CAPPA -DOPENMP -DSPMD -DUSE_COND -DUSE_GFSL63 -Duse_WRTCOMP -I/lfs4/NAGAPE/hpc-wof1/mansell/ufs/ufs-100-lr/src/ufs-stock/FV3/atmos_cubed_sphere -I/lfs4/NAGAPE/hpc-wof1/mansell/ufs/ufs-100-lr/src/ufs-stock/FV3/atmos_cubed_sphere/tools -I/lfs4/NAGAPE/hpc-wof1/mansell/ufs/ufs-100-lr/src/ufs-stock/build/FV3/ccpp/mod -I/lfs4/NAGAPE/hpc-wof1/mansell/ufs/ufs-100-lr/src/ufs-stock/build/FV3/ccpp/framework/src -I/lfs4/NAGAPE/hpc-wof1/mansell/ufs/ufs-100-lr/src/ufs-stock/build/FV3/ccpp/physics -I/lfs4/HFIP/hfv3gfs/nwprod/hpc-stack/libs/intel-18.0.5.274/impi-2018.4.274/fms/2021.03/include_r8 -I/lfs4/HFIP/hfv3gfs/nwprod/hpc-stack/libs/intel-18.0.5.274/impi-2018.4.274/netcdf/4.7.4/include -I/lfs4/HFIP/hfv3gfs/nwprod/hpc-stack/libs/intel-18.0.5.274/w3nco/2.4.1/include_d -I/lfs4/HFIP/hfv3gfs/nwprod/hpc-stack/libs/intel-18.0.5.274/impi-2018.4.274/esmf/8_2_0/mod -I/lfs4/HFIP/hfv3gfs/nwprod/hpc-stack/libs/intel-18.0.5.274/impi-2018.4.274/esmf/8_2_0/include -g -traceback -fpp -fno-alias -auto -safe-cray-ptr -ftz -assume byterecl -nowarn -sox -align array64byte -qno-opt-dynamic-align -real-size 64 -O2 -debug minimal -fp-model source -qoverride-limits -qopt-prefetch=3 -no-prec-div -no-prec-sqrt -axSSE4.2,CORE-AVX2 -g -traceback -fpp -fno-alias -auto -safe-cray-ptr -ftz -assume byterecl -nowarn -sox -align array64byte -qno-opt-dynamic-align -real-size 64 -no-prec-div -no-prec-sqrt -axSSE4.2,CORE-AVX2 -O2 -debug minimal -fp-model source -qoverride-limits -qopt-prefetch=3 -module include/fv3 -fPIC -qopenmp -c /lfs4/NAGAPE/hpc-wof1/mansell/ufs/ufs-100-lr/src/ufs-stock/FV3/atmos_cubed_sphere/driver/fvGFS/atmosphere.F90 -o CMakeFiles/fv3.dir/driver/fvGFS/atmosphere.F90.o
Using 8-byte addressing
Using pure routines.
Using allocatable derived type array members.
Using cray pointers.
/lfs4/NAGAPE/hpc-wof1/mansell/ufs/ufs-100-lr/src/ufs-stock/FV3/atmos_cubed_sphere/driver/fvGFS/atmosphere.F90(475): catastrophic error: **Internal compiler error: internal abort** Please report this error along with the circumstances in which it occurred in a Software Problem Report.  Note: File and line given may not be explicit cause of this error.

The atmos_cubed_sphere is at
commit fa86482e48f1d5b594acb369e68b8488de84dc66 (HEAD, origin/dev/emc)
Author: laurenchilutti <60401591+laurenchilutti@users.noreply.github.com>
Date:   Tue Nov 30 07:42:42 2021 -0500

Any ideas?

-- Ted Mansell (NOAA/NSSL)

**Comment by climbfuji** [comment]
@MicroTed Can you please paste the *exact* compile command you are using?

**Comment by MicroTed** [comment]
I pasted output from 'make VERBOSE=1' and the compile command starts on the second line with

/apps/intel/compilers_and_libraries_2018.5.274/linux/mpi/intel64/bin/mpiifort -DENABLE_QUAD_PRECISION [etc]

It is entirely possible there is something odd in my setup that is causing an issue, but I've compiled ufs a number of times without issue. I do a 'module purge' before loading the ufs_jet.intel module.

**Comment by climbfuji** [comment]
I meant the command you type for compiling the code, sorry for the confusion. For example:
```
./compile.sh jet.intel ...
```

**Comment by MicroTed** [comment]
Oh, sorry! For this test I just set the environment/modules and run:

module purge
module use modulefiles/
module load ufs_jet.intel
module load miniconda3
setenv CCPP_SUITES FV3_GFS_v15p2
setenv CMAKE_FLAGS -DAPP=ATM
./build.sh >& out &

**Comment by climbfuji** [comment]
Alright, you are hitting an known problem here. Because of the way your flags are set, the choice for the CCPP suite doesn't make it into the cmake build command and as a result:
```
Calling CCPP code generator (ccpp_prebuild.py) for all available suites ...
```
See the issue here: https://github.com/ufs-community/ufs-weather-model/issues/898. What you want to do is:
```
export CMAKE_FLAGS='-DAPP=ATM -DCCPP_SUITES=FV3_GFS_v15p2'
```
because then you get:
```
Calling CCPP code generator (ccpp_prebuild.py) for suites --suites=FV3_GFS_v15p2 ...
```

**Comment by MicroTed** [comment]
Ha, I knew it had to be my fault. Thanks so much for the help!

**Comment by climbfuji** [comment]
No problem. But if there is something wrong in the documentation, then it should be fixed.

**Comment by junwang-noaa** [comment]
@climbfuji Thanks for answering the question. I saw the build.sh section in  the wiki documentation was updated. I updated the last section of feeding build options to compile.sh.

https://github.com/ufs-community/ufs-weather-model/wiki/Building-model

@MicroTed Please let us know if the issue is resolved.

**Comment by MicroTed** [comment]
Ah, so it is then building for *all* suites, which also explains why it takes so long to finish. I just got it successfully compiled on jet with a correct setting for the suite. For previous compiles, I had edited build.sh to set up the suites, but this time was trying to run it "as intended." (Which I didn't quite manage, haha)

@climbfuji : In other news, I got gcc-10 and hpc-stack working on my mac setup, so that is back in business. (except eckit, but not needed)

RESOLVED!
